woodwork/statistics_utils.py

```diff
@@ -343,6 +343,136 @@ def _get_value_counts(dataframe, ascending=False, top_n=10, dropna=False):
+            - high_indices (list[int]): the corresponding index values for each of the upper outliers
+    """
+    if dd and isinstance(series, dd.Series):
+        series = series.compute()
+    if ks and isinstance(series, ks.Series):
+        series = series.to_pandas()
+
+    series = series.dropna()
+    if series.shape[0] == 0:
+        return
+
+    if quantiles is None:
+        quantiles = series.quantile([0.0, 0.25, 0.5, 0.75, 1.0]).to_dict()
+
+    if 0.25 in quantiles and 0.75 in quantiles:
+        low_bound, high_bound = _calculate_iqr_bounds(quantiles=quantiles)
+    else:
+        # if first and third quantile are not present, we have to calulate from the data
+        low_bound, high_bound = _calculate_iqr_bounds(series=series)
+
+    min = quantiles.get(0.0)
+    max = quantiles.get(1.0)
+    if min is not None and max is not None and low_bound <= min and high_bound >= max:
+        outliers_dict = {
+            "low_values": [],
+            "high_values": [],
+            "low_indices": [],
+            "high_indices": []
+        }
+    else:
+        # We've already removed nans and converted to pandas
+        outliers_dict = _get_outliers_for_column(series, low_bound, high_bound, convert_series=False)
+
+    return {'low_bound': low_bound,
+            'high_bound': high_bound,
+            'quantiles': quantiles,
+            **outliers_dict}
+
+
+def _get_outliers_for_column(series, low_bound=None, high_bound=None, convert_series=True):
+    """Gets the values of a series that lay above and below specified bounds.
+
+    Args:
+        series (Series): Data for which the outliers should be determined.
+        low_bound (float, optional): The number below which outliers lay. Is inclusive.
+        high_bound (float, optional): The number above which outliers lay. Is inclusive.
+
+    Note:
+        If neither or only one of low_bound or high_bound is passed in, the bounds will be calculated
+        using the IQR method.
+
+    Returns:
+        dict[str -> float,list[number]]: a dictionary containing outlier values and their corresponding indexes.
+            The following elements will be found in the dictionary:
+
+            - low_values (list[float, int]): the values of the lower outliers
+            - high_values (list[float, int]): the values of the upper outliers
+            - low_indices (list[int]): the corresponding index values for each of the lower outliers
+            - high_indices (list[int]): the corresponding index values for each of the upper outliers
+    """
+    if convert_series:
+        if dd and isinstance(series, dd.Series):
+            series = series.compute()
+        if ks and isinstance(series, ks.Series):
+            series = series.to_pandas()
+
+        series = series.dropna()
+        if series.shape[0] == 0:
+            return
+
+    if low_bound is None or high_bound is None:
+        low_bound, high_bound = _calculate_iqr_bounds(series)
```

Comment:
I agree that it can be confusing. Part of why I didn't want to allow partial inputs with the bounds (or with `Q1` and `Q3` down on line 406) was to avoid the added complexity that would come with being able to handle any combination of inputs. It also seems like a big assumption to have one bound be a user input and have the other bound be from the IQR.

Right now, I'm thinking it would be better to avoid these assumptions and raise errors if a user has passed in partial inputs, letting them know that they either need to pass in all the required stats or remove them all and let Woodwork calculate them.

Another thought I had is that we might want to catch a situation where a user has passed in bounds where `low_bound > high_bound`.

Reply:
> Right now, I'm thinking it would be better to avoid these assumptions and raise errors if a user has passed in partial inputs, letting them know that they either need to pass in all the required stats or remove them all and let Woodwork calculate them.

Yeah, I like your suggestion here.

Reply:
In removing the separate `outliers_dict` call, there is no longer a path that allows users to pass in bounds and get the values outside of the bounds. This situation is no longer possible, as we only care now about whether the correct quantiles were passed in.

How we handle partial quantiles is now also different:
- if there is a quantiles dict passed but q1 and q3 (0.25 and 0.75) are not present, then a `ValueError` is thrown
- if no quantiles dict is passed, the quantiles are generated
- if the correct quantiles are passed, they are used to calculate IQR